examples/3d_examples/MagneticCiliaCarpetCase/immersed_magnetic_cilia_carpet.py

```diff
@@ -171,9 +190,26 @@ def update_carpet_lag_grid_fields():
                 f"grid deviation L2 error: {grid_dev_error:.6f}"
             )
 
+        # Save averaged vorticity field
+        if period_timer >= period_timer_limit:
+            period_timer = 0.0
+            if save_data:
+                avg_io.save(
+                    h5_file_name=f"avg_flow_{no_period}.h5",
+                    time=flow_sim.time,
+                )
+
+            avg_vorticity *= 0.0
+            avg_velocity *= 0.0
+            no_period += 1
+
         # compute timestep
         flow_dt = flow_sim.compute_stable_timestep(dt_prefac=0.25)
 
+        # Average vorticity field
+        avg_vorticity += flow_sim.vorticity_field * flow_dt / period_timer_limit
+        avg_velocity += flow_sim.velocity_field * flow_dt / period_timer_limit
```

Comment:
Can you use an elementwise operation for this from pystencils? check out the saxpby kernel

Reply:
[fbd5513](https://github.com/SophT-Team/SophT/pull/120/commits/fbd5513ae66ab43926856f754aa07bef09e3ed80)